src/io/mod.rs

```diff
@@ -171,8 +171,16 @@ impl<D: DeserializeOwned> DelimFileReader<D> {
             .has_headers(true)
             .quoting(quote)
             .from_reader(reader);
-        assert!(csv_reader.has_headers(), "Expected input file to have a header row");
```

Comment:
I removed this assert. It wasn't actually checking for a header line but instead just checks that the reader is configured to read a header.